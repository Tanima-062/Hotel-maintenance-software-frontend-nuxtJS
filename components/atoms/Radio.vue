<template>
  <label class="inline-flex items-center">
    <input
      type="radio"
      :name="name"
      class="form-radio h-4 w-4 mr-1"
      :checked="checked"
      :value="value"
      @click="click(value)"
      :disabled="isDisabled"
    />
    <span :class="`${checked ? 'text-checkboxColor' : color}`">{{ text }}</span>
  </label>
</template>
<script lang="ts">
import { Vue, Component, Prop, Emit } from 'nuxt-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: '' })
  name!: string

  @Prop({ default: '' })
  value!: string | number

  @Prop({ default: '' })
  text!: string

  @Prop({ default: 'default' })
  color!: string

  @Prop({ default: false })
  checked!: boolean

  @Prop({ default: false })
  isDisabled!: boolean

  @Emit()
  private click(value: string | number): string | number | void {
    if (this.isDisabled) return
    return value
  }
}
</script>
