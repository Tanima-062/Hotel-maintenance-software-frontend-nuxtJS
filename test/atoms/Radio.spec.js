import { shallowMount } from '@vue/test-utils'
import Radio from '@/components/atoms/Radio.vue'

describe('Atoms Radio Check', () => {
  test('Radio Test', () => {
    const wrapper = shallowMount(Radio)

    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
