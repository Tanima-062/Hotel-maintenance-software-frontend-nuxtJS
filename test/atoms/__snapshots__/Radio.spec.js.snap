// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms Radio Check Radio Test 1`] = `
<label
  class="inline-flex items-center"
>
  <input
    class="form-radio h-4 w-4 mr-1"
    name=""
    type="radio"
    value=""
  />
   
  <span
    class="default"
  >
    
  </span>
</label>
`;
